FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,35 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard self != nil else { return }
+
+			switch result {
+			case let .success((data, httpURLResponse)):
+				if httpURLResponse.statusCode == 200,
+				   let remoteFeedImages = RemoteFeedLoader.parse(data: data) {
+					completion(.success(RemoteFeedLoader.map(remoteFeedImages)))
+				} else {
+					completion(.failure(Error.invalidData))
+				}
```

Comment:
Since checking the response `statusCode` and `parsing` data are parts of mapping logic in this context, they can be moved to the `map` function. By changing the `map` function type to `(Data, HTTPURLResponse) -> FeedLoader.Result` we can simplify the `load` logic:
```suggestion
				completion(RemoteFeedLoader.map(data, from: httpURLResponse))
```